app/main.py

```diff
@@ -16,4 +16,32 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def filter_variables(variables: dict) -> dict:
```

Comment:
you don't need function for this task, you can do it like this:

sorted_variables = {
    "mutable": [my_favourite_films, marks, collection_of_coins],
    "immutable": [lucky_number, pi, one_is_a_prime_number, name, profile_info]
}